src/supautils.c

```diff
@@ -332,7 +333,7 @@ supautils_hook(PROCESS_UTILITY_PARAMS)
 				foreach(option_cell, stmt->options)
 				{
 					DefElem *defel = (DefElem *) lfirst(option_cell);
-					if (strcmp(defel->defname, "superuser") == 0) {
+					if (strcmp(defel->defname, "superuser") == 0 && defGetBoolean(defel)) {
```

Comment:
This would allow `postgres` to downgrade `supabase_admin` to non-superuser

Reply:
Wouldn't it be blocked because `supabase_admin` is a privileged role? Similar to this attempt to "downgrade" `postgres` role to non-su https://github.com/supabase/supautils/pull/87/files#diff-d3dfe6cddc8ca24702b4ba1f9960ed1746a87993b0379c0a1e2cc9e6a3744dc3R187-R189

Reply:
Sure, but it's still incorrect logic-wise - `supautils.privileged_role` shouldn't be able to downgrade any superusers.

Reply:
I think it's sufficient to say we don't allow users to upgrade to superuser. After all, the main fix here is to revert the regression that was unintentionally introduced.

Reply:
Disagree, we can't rely on the role having to be in `reserved_roles`, that's just going to bite us down the line

Reply:
Why can't we rely on `reserved_roles`? That sounds like a separate RFC tbh. It shouldn't block the regression fix.

Reply:
The idea is to be secure and conservative by default - "privileged roles cannot modify superusers" is a useful invariant to have. We only add capabilities *if* and *when* users need them.

All we need to do is remove the change in this line.